providers/CollectionProvider.tsx

```diff
@@ -0,0 +1,53 @@
+import { COLLECTION_TYPE } from '@/types/collection'
+import { useQuery } from '@tanstack/react-query'
+import { useRouter } from 'next/navigation'
+import { createContext, useContext, useEffect, useMemo, useState } from 'react'
+import { useAccount } from 'wagmi'
+
+const CollectionContext = createContext(null)
+
+const CollectionProvider = ({ children }) => {
+  const { address } = useAccount()
+  const [selectedCollection, setSelectedCollection] = useState<COLLECTION_TYPE | null>(null)
+  const { push } = useRouter()
+  const getCollections = async (): Promise<COLLECTION_TYPE[]> => {
+    const response = await fetch(`/api/collections?address=${address}`)
+    const data = await response.json()
+    return data.collections
+  }
+
+  const { data: collections } = useQuery({
+    queryKey: ['getCollections'],
+    queryFn: getCollections,
+  })
+
+  useEffect(() => {
+    if (selectedCollection) {
+      push(`/${selectedCollection.chainId}/${selectedCollection.address}`)
+      return
+    }
+
+    push('/')
+  }, [selectedCollection])
+
+  const value = useMemo(
+    () => ({
+      collections,
+      setSelectedCollection,
+      selectedCollection,
+    }),
+    [collections, setSelectedCollection, selectedCollection],
+  )
```

Comment:
optional: move all hooks / state out of the provider and into a standalone `hook`.
- rule: each file should do exactly one thing.